2004-09-17  Gwenole Beauchesne <gbeauchesne@mandrakesoft.com>

	* Make-lang.in (bldtools): add enough .NOT_PARALLEL to let it
	build with GNU make -j4.

2004-09-01  Laurent GUERBY <laurent@guerby.net>

	* Make-lang.in (bldtools): force rebuild of
	runtime with no optimization for bltools to workaround
	a code generation bug in 3.4.2.

--- gcc-4.0.1/gcc/ada/Make-lang.in.ada-64bit-hack	2005-03-16 09:38:31.000000000 -0500
+++ gcc-4.0.1/gcc/ada/Make-lang.in	2005-05-04 06:06:08.855029013 -0400
@@ -877,30 +877,30 @@ ada/b_gnatb.c : $(GNATBIND_OBJS) ada/gna
 	$(GNATBIND) -C $(ADA_INCLUDES) -o ada/b_gnatb.c ada/gnatbind.ali
 ada/b_gnatb.o : ada/b_gnatb.c
 
-ada/treeprs.ads : ada/treeprs.adt ada/sinfo.ads ada/xtreeprs.adb
+.NOT_PARALLEL ada/treeprs.ads : ada/treeprs.adt ada/sinfo.ads ada/xtreeprs.adb
 	-$(MKDIR) ada/bldtools/treeprs
 	$(CP) $^ ada/bldtools/treeprs
-	(cd ada/bldtools/treeprs; gnatmake -q xtreeprs ; ./xtreeprs ../../treeprs.ads )
+	(cd ada/bldtools/treeprs; gnatmake -q -a -f xtreeprs ; ./xtreeprs ../../treeprs.ads )
 
-ada/einfo.h : ada/einfo.ads ada/einfo.adb ada/xeinfo.adb
+.NOT_PARALLEL ada/einfo.h : ada/einfo.ads ada/einfo.adb ada/xeinfo.adb
 	-$(MKDIR) ada/bldtools/einfo
 	$(CP) $^ ada/bldtools/einfo
-	(cd ada/bldtools/einfo; gnatmake -q xeinfo ; ./xeinfo ../../einfo.h )
+	(cd ada/bldtools/einfo; gnatmake -q -a -f xeinfo ; ./xeinfo ../../einfo.h )
 
-ada/sinfo.h : ada/sinfo.ads ada/xsinfo.adb
+.NOT_PARALLEL ada/sinfo.h : ada/sinfo.ads ada/xsinfo.adb
 	-$(MKDIR) ada/bldtools/sinfo
 	$(CP) $^ ada/bldtools/sinfo
-	(cd ada/bldtools/sinfo; gnatmake -q xsinfo ; ./xsinfo ../../sinfo.h )
+	(cd ada/bldtools/sinfo; gnatmake -q -a -f xsinfo ; ./xsinfo ../../sinfo.h )
 
-ada/nmake.adb : ada/sinfo.ads ada/nmake.adt ada/xnmake.adb
+.NOT_PARALLEL ada/nmake.adb : ada/sinfo.ads ada/nmake.adt ada/xnmake.adb
 	-$(MKDIR) ada/bldtools/nmake_b
 	$(CP) $^ ada/bldtools/nmake_b
-	(cd ada/bldtools/nmake_b; gnatmake -q xnmake ; ./xnmake -b ../../nmake.adb )
+	(cd ada/bldtools/nmake_b; gnatmake -q -a -f xnmake ; ./xnmake -b ../../nmake.adb )
 
-ada/nmake.ads :  ada/sinfo.ads ada/nmake.adt ada/xnmake.adb ada/nmake.adb
+.NOT_PARALLEL ada/nmake.ads :  ada/sinfo.ads ada/nmake.adt ada/xnmake.adb ada/nmake.adb
 	-$(MKDIR) ada/bldtools/nmake_s
 	$(CP) $^ ada/bldtools/nmake_s
-	(cd ada/bldtools/nmake_s; gnatmake -q xnmake ; ./xnmake -s ../../nmake.ads )
+	(cd ada/bldtools/nmake_s; gnatmake -q -a -f xnmake ; ./xnmake -s ../../nmake.ads )
 
 update-sources : ada/treeprs.ads ada/einfo.h ada/sinfo.h ada/nmake.adb \
 	ada/nmake.ads
